### ISSUE TITLE
TUploadfile is causing a memory leak?

### ISSUE THREAD

**Issue by cvezina** [QuickBlox/quickblox-ios-sdk#387]
Doing a simple call to TUploadFile is causing a memory leak. Basically, if I am in a QBChatDialog and the user send a message that is an image (via TUploadFile), then after the user leave the chat dialog, dealloc of the object encapsulating the QBChatDialog is not called. If no image are sent (just text messages), everything is working as expected and when the user leave the dialog, dealloc is called as expected. 

Here is the sample code used:

```objectivec
- (void) sendImage: (UIImage*) image
{
	NSData *imageData = UIImageJPEGRepresentation (image, 0.3);
	
	[self startLoading];

	// TODO: TUploadFile is causing a leak
	
	[QBRequest TUploadFile:imageData fileName:@"image.jpeg" contentType:@"image/jpg" isPublic: NO successBlock:^(QBResponse *response, QBCBlob *uploadedBlob) {
	} statusBlock:^(QBRequest *request, QBRequestStatus *status) {
		[self stopLoading];
	} errorBlock:^(QBResponse *response) {
		[self stopLoading];
		[[TWMessageBarManager sharedInstance] showMessageWithTitle: @"Something went wrong!" description: nil type: TWMessageBarMessageTypeError];
		NSLog(@"GalPal - (Error) : Error while sending image: %@", response.error.description);
	}];
}
```

**Comment by Raikerian** [comment]
Hi @cvezina 

Thank you for reporting the issue. I was able to catch and fix it. It will be available in the next release.
For now there is a possible solution for you to make a weak reference for self, in order for block not to retain it. Here is an example:

```objectivec
- (void) sendImage: (UIImage*) image
{
    NSData *imageData = UIImageJPEGRepresentation (image, 0.3);
    
    [self startLoading];
    
    // TODO: TUploadFile is causing a leak
    
    __weak __typeof(self)weakSelf = self;
    [QBRequest TUploadFile:imageData fileName:@"image.jpeg" contentType:@"image/jpg" isPublic: NO successBlock:nil
               statusBlock:^(QBRequest *request, QBRequestStatus *status) {
                   
        [weakSelf stopLoading];
        
    } errorBlock:^(QBResponse *response) {
        
        [weakSelf stopLoading];
        [[TWMessageBarManager sharedInstance] showMessageWithTitle: @"Something went wrong!" description: nil type: TWMessageBarMessageTypeError];
        NSLog(@"GalPal - (Error) : Error while sending image: %@", response.error.description);
    }];
}
```

**Comment by Raikerian** [comment]
Fixed in 2.7.1.
